pkg/controller/volume/attachdetach/cache/desired_state_of_world.go

```diff
@@ -318,6 +351,18 @@ func (dsw *desiredStateOfWorld) NodeExists(nodeName k8stypes.NodeName) bool {
 	return nodeExists
 }
 
+func (dsw *desiredStateOfWorld) SafeToDetach(nodeName k8stypes.NodeName) bool {
+
+	// do we need lock here?
```

Comment:
could someone help here, do we need lock in this function or not?

Reply:
I think lock is needed here